src/block-ciphers.rst

```diff
@@ -12,109 +12,103 @@ Block ciphers
-of those blocks maps to a hexadecimal digit, we'll represent the blocks
+encryption and decryption.
+
+A block cipher is a *keyed permutation*. First, the key determines exactly which
+blocks are mapped. Second, the permutation aspect allows mapping of all possible blocks 
+to other blocks. Permutation is important because the recipient must map blocks back to the 
+original blocks by analyzing one-to-one.
+
+We illustrate this by looking at a block cipher with an impractical,
+tiny 4-bit block size. :math:`2^4 = 16` possible blocks. Since each
+of the blocks map to a hexadecimal digit, we represent the blocks
 by that digit. :numref:`fig-BlockCipherBlocks`
-illustrates the blocks that the cipher operates on.
+illustrates blocks that the cipher operates on.
 
 
 .. _fig-BlockCipherBlocks:
 
 .. figure:: Illustrations/BlockCipher/AllNodes.svg
    :align: center
 
-   All of the 16 nodes operated on by the block cipher. Each node is designated by a hexadecimal digit.
+   All 16 nodes operated on by the block cipher. Each node is designated by a hexadecimal digit.
 
-Once we select a secret key, the block cipher will use that to determine
-what the encryption of any given block is. We will illustrate that
-relationship with an arrow: the block at the start of the arrow,
-encrypted using :math:`E` under key :math:`k`, is mapped to the block at
-the end of the arrow.
+Once we select a secret key, the block cipher uses it to determine
+the encryption of any given block. We illustrate that
+relationship with an arrow. The tail of the arrow has the block
+encrypted with :math:`E` under key :math:`k` and the arrowhead is mapped to the block.
 
 .. _fig-BlockCipherEncryption:
 
 .. figure:: Illustrations/BlockCipher/Encryption.svg
    :align: center
 
-   An encryption permutation produced by the block cipher under a particular key :math:`k`.
+   An encryption permutation made by a block cipher under a particular key :math:`k`.
 
-In :numref:`fig-BlockCipherEncryption`, you'll notice
-that the permutation isn't just one big cycle: there's a large cycle of
-7 elements, and several smaller cycles of 4, 3 and 2 elements each. It's
+In :numref:`fig-BlockCipherEncryption`, note
+that the permutation is not just one big cycle. It contains a large cycle of
+7 elements, and several smaller cycles of 4, 3 and 2 elements each. It is
 also perfectly possible that an element encrypts to itself. This is to
 be expected when selecting random permutations, which is approximately
 what a block cipher is doing; it doesn't demonstrate a bug in the block
 cipher.
 
-When you're decrypting instead of encrypting, the block cipher just
+When you decrypt instead of encrypt, the block cipher 
 computes the inverse permutation. In :numref:`fig-BlockCipherDecryption`,
-you can see that we get the same illustration, except that all the arrows are
-going in the other direction.
+we get the same illustration. The exception is that all arrowheads point
```

Comment:
This seems to imply that it's only accidentally the case that the arrowheads are reversed, but that's a fundamental property. Perhaps "The difference is" or "The difference between the illustrations is"?